glib-macros/tests/properties.rs

```diff
@@ -0,0 +1,221 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::prelude::*;
+use glib::ParamFlags;
+#[test]
+fn props() {
+    mod foo {
+        use glib::prelude::*;
+        use glib::subclass::prelude::*;
+        use glib_macros::Properties;
+        use std::cell::RefCell;
+        use std::marker::PhantomData;
+        use std::sync::Mutex;
+
+        use once_cell::sync::OnceCell;
+
+        #[derive(Clone, Default, Debug, PartialEq, Eq, glib::Boxed)]
+        #[boxed_type(name = "SimpleBoxedString")]
+        pub struct SimpleBoxedString(pub String);
+
+        #[derive(Default, Clone)]
+        struct Author {
+            name: String,
+            nick: String,
+        }
+
+        pub mod imp {
+            use glib::{ParamSpec, Value};
+            use std::rc::Rc;
+
+            use super::*;
+
+            #[derive(Properties, Default)]
+            pub struct Foo {
+                #[prop(get, set)]
+                bar: Mutex<String>,
+                #[prop(get = Self::hello_world)]
+                _buzz: PhantomData<String>,
+                #[prop(get, set = Self::set_fizz, name = "fizz")]
+                fizz: RefCell<String>,
+                #[prop(name = "author-name", get, set, type = String, member = name)]
+                #[prop(name = "author-nick", get, set, type = String, member = nick)]
+                author: RefCell<Author>,
+                #[prop(
+                    type = String,
+                    get = |t: &Self| t.author.borrow().name.to_owned(),
+                    set = Self::set_author_name)]
+                fake_field: PhantomData<String>,
+                #[prop(get, set, user_1, user_2, lax_validation)]
+                custom_flags: RefCell<String>,
+                #[prop(get, set, builder())]
+                simple_builder: RefCell<u32>,
+                #[prop(get, set, builder().minimum(0).maximum(5))]
+                numeric_builder: RefCell<u32>,
+                #[prop(get, set, builder('c'))]
+                builder_with_required_param: RefCell<char>,
+                #[prop(get, set, builder(SimpleBoxedString::static_type()))]
+                boxed: RefCell<SimpleBoxedString>,
+                #[prop(get, set)]
+                optional: RefCell<Option<String>>,
+                #[prop(get, set)]
+                smart_pointer: Rc<RefCell<String>>,
+                #[prop(get, set)]
+                once_cell: OnceCell<u8>,
+            }
```

Comment:
Heya, @ranfdev! The example and test code doesn't seem to include any examples to set the description of the fields. Can you add them, please?

Reply:
Check out the latest commit and feel free to add feedback.